NCS.DSS.EmploymentProgression/PatchEmploymentProgression/Function/EmploymentProgressionPatchTrigger.cs

```diff
@@ -0,0 +1,225 @@
+                                                )]
+        public async Task<HttpResponseMessage> Run([HttpTrigger(AuthorizationLevel.Anonymous, "patch", Route = RouteValue)]HttpRequest req, ILogger logger, string customerId, string EmploymentProgressionId)
+        {
+            _loggerHelper.LogMethodEnter(logger);
+
+            var correlationId = _httpRequestHelper.GetDssCorrelationId(req);
+
+            var guidHelper = new GuidHelper();
+
+            var correlationGuid = guidHelper.ValidateGuid(correlationId);
+
+            var touchpointId = _httpRequestHelper.GetDssTouchpointId(req);
+            if (string.IsNullOrEmpty(touchpointId))
+            {
+                _loggerHelper.LogInformationMessage(logger, correlationGuid, "Unable to locate 'TouchpointId' in request header.");
+
+                return _httpResponseMessageHelper.BadRequest();
+            }
+
+            var ApimURL = _httpRequestHelper.GetDssApimUrl(req);
+            if (string.IsNullOrEmpty(ApimURL))
+            {
+                _loggerHelper.LogInformationMessage(logger, correlationGuid, "Unable to locate 'apimurl' in request header");
+                return _httpResponseMessageHelper.BadRequest();
+            }
+
+            if (!Guid.TryParse(customerId, out var customerGuid))
+            {
+                _loggerHelper.LogInformationMessage(logger, correlationGuid, $"Unable to parse 'customerId' to a Guid: {customerId}");
+                return _httpResponseMessageHelper.BadRequest(customerGuid);
+            }
+
+            if (!Guid.TryParse(EmploymentProgressionId, out var employmentProgressionGuid))
+            {
+                _loggerHelper.LogInformationMessage(logger, correlationGuid, $"Unable to parse 'employmentProgressionId' to a Guid: {employmentProgressionGuid}");
+                return _httpResponseMessageHelper.BadRequest(employmentProgressionGuid);
+            }
+
+            EmploymentProgressionPatch employmentProgressionPatchRequest;
+
+            try
+            {
+                employmentProgressionPatchRequest = await _httpRequestHelper.GetResourceFromRequest<EmploymentProgressionPatch>(req);
+            }
+            catch (Exception ex)
+            {
+                _loggerHelper.LogException(logger, correlationGuid, "Unable to retrieve body from req", ex);
+                return _httpResponseMessageHelper.UnprocessableEntity(JObject.FromObject(new { Error = ex.Message }).ToString());
+            }
+
+            if (employmentProgressionPatchRequest == null)
+            {
+                _loggerHelper.LogInformationMessage(logger, correlationGuid, $"A patch body was not provided. CorrelationId: {correlationGuid}.");
+                return _httpResponseMessageHelper.NoContent();
+            }
+
+            _employmentProgressionPatchTriggerService.SetIds(employmentProgressionPatchRequest, employmentProgressionGuid, touchpointId);
```

Comment:
not consistent with the other 14 resources.  Model can set the Id's

Reply:
Agreed to keep functionality in Service.